src/Common/Auth/AuthUtils.php

```diff
@@ -0,0 +1,652 @@
+            if (empty($hash)) {
+                error_log('OpenEMR Error : OpenEMR is not working because broken function.');
+                die("OpenEMR Error : OpenEMR is not working because broken function.");
+            }
+
+            // Set up session environment
+            $_SESSION['authUser'] = $username;
+            $_SESSION['authPass'] = $hash;
+            $_SESSION['authGroup'] = $authGroup['name'];
+            $_SESSION['authUserID'] = $userInfo['id'];
+            $_SESSION['authProvider'] = $authGroup['name'];
+            $_SESSION['authId'] = $userInfo['id'];
+            $_SESSION['userauthorized'] = $userInfo['authorized'];
+            // Some users may be able to authorize without being providers:
+            if ($userInfo['see_auth'] > '2') {
+                $_SESSION['userauthorized'] = '1';
+            }
+            EventAuditLogger::instance()->newEvent('login', $username, $authGroup['name'], 1, "success: " . $ip['ip_string']);
+        } else if ($this->apiAuth) {
+            // Set up class variables that the api will need to collect (log for API is done outside)
+            $this->userId = $userInfo['id'];
+            $this->userGroup = $authGroup['name'];
+        } else {
+            // Log for authentication that are done, which are not api auth or login auth
+            EventAuditLogger::instance()->newEvent('auth', $username, $authGroup['name'], 1, "Auth success: " . $ip['ip_string']);
+        }
+        return true;
+    }
+
+    /**
+     * Setup or change a user's password
+     *
+     * @param type $activeUser      ID of who is trying to make the change (either the user himself, or an administrator) - CAN NOT BE EMPTY
+     * @param type $targetUser      ID of what account's password is to be updated (for a new user this doesn't exist yet).
+     * @param type $currentPwd      the active user's current password - CAN NOT BE EMPTY
+     *                              - password is passed by reference so that it can be "cleared out" as soon as we are done with it.
+     * @param type $newPwd          the new password for the target user
+     *                              - password is passed by reference so that it can be "cleared out" as soon as we are done with it.
+     * @param type $errMsg          passed by reference to return any
+     * @param type $create          Are we creating a new user or
+     * @param type $insert_sql      SQL to run to create the row in "users" (and generate a new id) when needed.
+     * @param type $new_username    The username for a new user
+     * @param type $newid           Return by reference of the ID of a created user
+     * @return boolean              Was the password successfully updated/created? If false, then $errMsg will tell you why it failed.
+     */
+    public function updatePassword($activeUser, $targetUser, &$currentPwd, &$newPwd, &$errMsg, $create = false, $insert_sql = "", $new_username = null, &$newid = null)
```

Comment:
TODO
Still need to convert this function fully to class (ie. migrate errMsg and newid to be stored in class rather than as paramaters).